Tone/instrument/CymbalSynth.js

```diff
@@ -0,0 +1,304 @@
+	 *                             see defaults below
+	 *  @example
+	 * var cymbal = new Tone.CymbalSynth().toMaster();
+	 * cymbal.harmonicity.value = 10;
+	 * cymbal.triggerAttack();
+	 */
+	Tone.CymbalSynth = function(options){
+
+		options = this.defaultArg(options, Tone.CymbalSynth.defaults);
+		Tone.Instrument.call(this, options);
+
+		/**
+		 * The initial strike component of the cymbal hit.
+		 * @type {CymbalComponent}
+		 */
+		this.strike = new CymbalComponent({
+			"volume" : options.strike.volume,
+			"cutoff" : options.strike.cutoff,
+			"frequencyScalar" : 1.886,
+			"resonance" : options.strike.resonance,
+			"resonanceScalar" : 0.533,
+			"envelope" : {
+				"attack": options.strike.envelope.attack, 
+				"decay": options.strike.envelope.decay, 
+			}
+		}).connect(this.output);
+
+		/**
+		 * The body portion of the cymbal hit.
+		 * @type {CymbalComponent}
+		 */
+		this.body = new CymbalComponent({
+			"volume" : options.body.volume,
+			"cutoff" : options.body.cutoff,
+			"frequencyScalar" : 1.643,
+			"resonance" : options.body.resonance,
+			"resonanceScalar" : 0.333,
+			"envelope" : {
+				"attack": options.body.envelope.attack, 
+				"decay": options.body.envelope.decay, 
+			}
+		}).connect(this.output);
+
+		/**
+		 *  [harmonicity description]
+		 *  @type {Positive}
+		 *  @signal
+		 */
+		this.harmonicity = new Tone.Signal(options.harmonicity);
+		this.harmonicity.units = Tone.Type.Positive;
+
+		/**
+		 *  The base frequency for the series of oscillators. 
+		 *  All oscillators scale in a harmonic or inharmonic pattern
+		 *  depending on the value of the harmonicity.
+		 *  @type {Frequency}
+		 *  @signal
+		 */
+		this.frequency = new Tone.Signal(options.frequency, Tone.Type.Frequency);
+
+		/**
+		 *  Inharmonic ratio of frequencies based on the Roland TR-808
+		 *  Taken from https://ccrma.stanford.edu/papers/tr-808-cymbal-physically-informed-circuit-bendable-digital-model
+		 *  @private
+		 *  @static
+		 *  @type {Array}
+		 */
+		this.inharmRatios = [1.0, 1.483, 1.932, 2.546, 2.630, 3.897];
+
+		/**
+		 *  Harmonic ratio that produces a more pitched cymbal
+		 *  @private
+		 *  @static
+		 *  @type {Array}
+		 */
+		this.harmRatios = [1.0, 1.5, 2.025, 2.975, 4.0, 6.0];
+
+		/**
+		 *  A series of FMSynths
+		 *  @private
+		 *  @type {Array}
+		 */
+		this._oscillators = [];
+
+		/**
+		 *  Tone.ScaledSignals used for interpolating between harmonic
+		 *  and inharmonic ratios
+		 *  @private
+		 *  @type {Array}
+		 */
+		this._scaledSignals = [];
+
+		/**
+		 *  Tone.Multipliers for frequency multiplication of scaled signals
+		 *  @private
+		 *  @type {Array}
+		 */
+		this._freqMult  = [];
+
+		//make all the FMSynths, connect the frequency signals
+		for(var i = 0; i < 6; i++){
+			this._scaledSignals[i] = new Tone.Scale(this.inharmRatios[i], this.harmRatios[i]);
+			this._freqMult[i] = new Tone.Multiply();
+			this._oscillators[i] = new Tone.SimpleFM({
```

Comment:
I'm still running into some issues with this not sounding the same, and weird things like signal values such as _freqMult[i] being zero when they should be the frequency value for each oscillator

```		/**
		 *  The level of inharmonicity between the frequencies of the oscillators. 
		 *  Interesting sounds can be had between 0 and 20.
		 *  @type {Positive}
		 *  @signal
		 */
		this.inharmonicity = new Tone.Signal(options.inharmonicity, Tone.Type.Positive);

		/**
		 *  The base frequency for the series of oscillators. 
		 *  All oscillators scale in a harmonic or inharmonic pattern
		 *  depending on the value of the inharmonicity.
		 *  @type {Frequency}
		 *  @signal
		 */
		this.frequency = new Tone.Signal(options.frequency, Tone.Type.Frequency);

		/**
		 *  A series of FMSynths
		 *  @private
		 *  @type {Array}
		 */
		this._carriers = [];

		this._modulators = [];

		/**
		 *  Tone.ScaledSignals used for interpolating between harmonic
		 *  and inharmonic ratios
		 *  @private
		 *  @type {Array}
		 */
		this._scaledSignals = [];

		/**
		 *  Tone.Multipliers for frequency multiplication of scaled signals
		 *  @private
		 *  @type {Array}
		 */
		this._freqMult  = [];

		this.harmonicity = [];

		this._modulationNodes = [];

		this._modulationIndex = [];

		//make all the FMSynths, connect the frequency signals
		for(var i = 0; i < 6; i++){
			this._scaledSignals[i] = new Tone.Scale(inharmRatios[i], harmRatios[i]);
			this._freqMult[i] = new Tone.Multiply();
			this._carriers[i] = new Tone.Oscillator();
			this._carriers[i].type = "square";
			this._carriers[i].volume.value = -10;
			
			this._modulators[i] = new Tone.OmniOscillator();
			this._modulators[i].type = "pulse";
			this._modulators[i].width.value = 0.75;
			this._modulators[i].volume.value = -10;

			this.harmonicity[i] = new Tone.Multiply(1.5);

			this._modulationIndex[i] = new Tone.Multiply(10);


			this.inharmonicity.connect(this._scaledSignals[i]);
			this._scaledSignals[i].connect(this._freqMult[i], 0, 0);
			this.frequency.connect(this._freqMult[i], 0, 1);

			this._modulationNodes[i] = this.context.createGain();
			this._modulators[i].connect(this._modulationNodes[i].gain);

			this._freqMult[i].chain(this.harmonicity[i], this._modulationNodes[i]);
			this._freqMult[i].chain(this._modulationIndex[i], this._modulationNodes[i]);
			this._modulationNodes[i].gain.value = 0;

			this._modulationNodes[i].connect(this._carriers[i].frequency);
			this._freqMult[i].connect(this._carriers[i].frequency);
			this._carriers[i].connect(this.strike);
			this._carriers[i].connect(this.body);

			this._carriers[i].start();
			this._modulators[i].start();
		}
	};

```

Here's what I have so far. I'm really stuck on debugging this. Any idea what I'm doing wrong?